packages/core/src/controller/object-controller.ts

```diff
@@ -0,0 +1,46 @@
+import { Task } from '../task';
+import { Controller } from './controller';
+import { createFuture } from '../future';
+import { OperationObject } from '../operation';
+
+export function createObjectController<TOut>(task: Task<TOut>, _obj: OperationObject<TOut>, createController: () => Controller<TOut>): Controller<TOut> {
+  let delegate: Controller<TOut>;
+  let { produce, future } = createFuture<TOut>();
+
+  function start() {
+    try {
+      delegate = createController();
+      task.setLabels({ ...delegate.operation?.labels });
```

Comment:
@cowboyd, what do you think about it? `setLabels` already merges `task.labels` and even more `task.labels == obj.labels` so I thought there is no reason to pass those.